sdk/span.go

```diff
@@ -10,9 +10,14 @@ type Attribute struct {
 	Value interface{}
 }
 
+type Iterator interface {
+	HasNext() bool
+	Next() Attribute
```

Comment:
@ryanericson @tim-mwangi Still returning the generic interface Attribute since we want to fetch the key as well. Or we can return the otel key-value pair directly

Reply:
Or we can keep on using the same `Attribute` object so that it's initialised only once

Reply:
Why not do something like: https://bitfieldconsulting.com/golang/iterators#:~:text=A%20Golang%20iterator%20is%20a,value%20returned%20by%20the%20iterator.

Reply:
But here also what would we pass as the argument of the yield function? If we want to keep the generality of the attribute, we would have to return a common struct right?
like 
```
func (l *AttributeList) IterateItems(yield func(attr Attribute) bool) {
	for _, attr := range l.attrs {
		if !yield(attr) {
			return
		}
	}
}
```

Reply:
Thanks, updated the pr. Didnt know we could make an iterator like this